src/status_im/transport/message/core.cljs

```diff
@@ -58,9 +58,8 @@
       (seq chats)
       (do
         (js-delete response-js "chats")
-        (rf/merge cofx
-                  (process-next response-js sync-handler)
-                  (chat.events/ensure-chats (map data-store.chats/<-rpc (types/js->clj chats)))))
+        {:fx [[:dispatch [:transport/process-next response-js sync-handler]]
```

Comment:
FYI: This shows how `fx` orders the effects. First `:transport/process-next` is scheduled (behind the scenes it will run ~20ms later because it uses `dispatch-later`), then `:chat/ensure-chats` will run. Remember both effects may run on the same batch depending on how busy re-frame is by the time this signal arrives, which is a good thing for responsiveness.

Reply:
I guess you might want to do the opposite here, although it doesn't matter so much, but I find it more natural to first process the event we currently care about (i.e processing chats), and later dispatching the `process next` event, seems more natural, but maybe that's just me

Reply:
It's not just you @cammellos, I had this question myself too. Either way works, but you have a point.

Reply:
Resolve in https://github.com/status-im/status-mobile/pull/17451/commits/c21bb0367a298ac28fa2688b0c1999f349e37287